### PR TITLE
openssl10: added to simplify migration to OpenSSL 1.1 

### PR DESCRIPTION
**Pull request by yan12125**
#### Description

openssl10: added to simplify migration to OpenSSL 1.1

Ref: https://trac.macports.org/ticket/52101
Ref: https://github.com/macports/macports-ports/pull/3822

The following python26 is an example for how to use the openssl10 port. test_ssl and test_hashlib still pass after the change.

The post-destroot is inspired by the Arch Linux package [1]. I don't move dylibs around as the linker on macOS uses absolute paths.

[1] https://git.archlinux.org/svntogit/packages.git/tree/trunk/PKGBUILD?h=packages/openssl-1.0#n67

###### Type(s)
- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on
macOS 10.14.3 18D109
Xcode 10.1 10B61

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?
- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: lang/python26/Portfile [hunk 1]
```diff
@@ -61,6 +61,8 @@ configure.ccache        no
 # signed integer overflow is normally undefined as per the C spec, but python
 # needs it to be defined as twos-complement
 configure.cflags-append -fwrapv
+configure.cppflags-append -I${prefix}/include/openssl-1.0
+configure.ldflags-append -L${prefix}/lib/openssl-1.0
```
Comment (jmroot) [inline lang/python26/Portfile]: Does this find the right headers and lib with both openssl10 and openssl11 installed? I would have thought it needed to be a prepend.

Comment (yan12125) [reply lang/python26/Portfile]: That makes sense, thanks! Changed.

Comment (yan12125) [reply lang/python26/Portfile]: Oops, for some unknown reason the latest changes are not commited. Should be correct now.

File: devel/openssl10/Portfile [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    move ${destroot}${prefix}/bin/openssl ${destroot}${prefix}/bin/openssl-1.0
+
+    # Update includedir in .pc files
+    reinplace "s|/include$|/include/openssl-1.0|" {*}[glob ${destroot}${prefix}/lib/openssl-1.0/pkgconfig/*.pc]
+
+    file delete -force ${destroot}${prefix}/etc
+    file delete ${destroot}${prefix}/bin/c_rehash
+}
+
+destroot.args       MANDIR=${prefix}/share/man MANSUFFIX=ssl
```
Comment (jmroot) [inline devel/openssl10/Portfile]: Won't the man pages conflict with the openssl port?

Comment (jmroot) [reply devel/openssl10/Portfile]: Never mind, I see that's taken care of by the destroot.target. I guess there's no need for these args though.

**Comment by mf2k** [comment]
It seems like this would be better kept in the openssl port and just made as a sub-port. Only a few lines are different.

File: devel/openssl10/Portfile (hunk 2 above)
Comment (yan12125) [reply devel/openssl10/Portfile]: Thank you! This line is now removed.

**Comment by yan12125** [comment]
> It seems like this would be better kept in the openssl port and just made as a sub-port. Only a few lines are different.

I wonder if current openssl maintainers want to care about this old version or not. I guess not given the comment https://github.com/macports/macports-ports/pull/3822#pullrequestreview-212565147.

**Comment by yan12125** [comment]
Gonna merge to move on the OpenSSL 1.1 update. Feel free to commit changes to this port. As I've indicated in the `maintainers` line of this port, I don't plan to maintain it.

**Pull request merged**